engine/src/test/java/org/camunda/bpm/engine/test/api/history/removaltime/batch/BatchSetRemovalTimeHierarchicalTest.java

```diff
@@ -702,6 +702,93 @@ public void shouldNotSetRemovalTime_HistoricInstancePermissionsDisabled() {
     testRule.clearAuthorization();
   }
 
+  @Test
+  public void shouldSetRemovalTime_HistoricProcessInstanceAuthorization() {
+    // given
+    testRule.getProcessEngineConfiguration()
+        .setEnableHistoricInstancePermissions(true);
+
+    String processInstanceId = testRule.process().call().userTask().deploy().start();
+
+    Authorization authorization =
+        authorizationService.createNewAuthorization(Authorization.AUTH_TYPE_GRANT);
+    authorization.setResource(Resources.HISTORIC_PROCESS_INSTANCE);
+    authorization.setResourceId(processInstanceId);
+    authorization.setUserId("foo");
+
+    authorizationService.saveAuthorization(authorization);
+
+    // assume
+    authorization = authorizationService.createAuthorizationQuery().singleResult();
+
+    assertThat(authorization.getRemovalTime()).isNull();
+
+    // when
+    testRule.updateHistoryTimeToLive("rootProcess", 5);
+
+    HistoricProcessInstanceQuery query =
+        historyService.createHistoricProcessInstanceQuery().rootProcessInstances();
+
+    testRule.syncExec(
+        historyService.setRemovalTimeToHistoricProcessInstances()
+            .calculatedRemovalTime()
+            .byQuery(query)
+            .hierarchical()
+            .executeAsync()
+    );
+
+    // then
+    authorization = authorizationService.createAuthorizationQuery()
+            .resourceType(Resources.HISTORIC_PROCESS_INSTANCE)
+            .singleResult();
+
+    assertThat(authorization.getRemovalTime()).isEqualTo(addDays(CURRENT_DATE, 5));
+
+    // clear
+    testRule.clearAuthorization();
+  }
+
+  @Test
+  public void shouldNotSetRemovalTime_HistoricProcessInstancePermissionsDisabled() {
+    // given
+    testRule.getProcessEngineConfiguration()
+        .setEnableHistoricInstancePermissions(false);
+
+    String processInstanceId = testRule.process().call().userTask().deploy().start();
+
+    Authorization authorization =
+        authorizationService.createNewAuthorization(Authorization.AUTH_TYPE_GRANT);
+    authorization.setResource(Resources.HISTORIC_PROCESS_INSTANCE);
+    authorization.setResourceId(processInstanceId);
+    authorization.setUserId("foo");
+
+    authorizationService.saveAuthorization(authorization);
+
+    testRule.updateHistoryTimeToLive("rootProcess", 5);
+
+    // when
+    HistoricProcessInstanceQuery query =
+        historyService.createHistoricProcessInstanceQuery().rootProcessInstances();
+
+    testRule.syncExec(
+        historyService.setRemovalTimeToHistoricProcessInstances()
+            .calculatedRemovalTime()
+            .byQuery(query)
+            .hierarchical()
+            .executeAsync()
+    );
+
+    // then
+    authorization = authorizationService.createAuthorizationQuery()
+        .resourceType(Resources.HISTORIC_PROCESS_INSTANCE)
+        .singleResult();
+
+    assertThat(authorization.getRemovalTime()).isNull();
+
+    // clear
+    testRule.clearAuthorization();
```

Comment:
Should go into `@After` method to make sure it clears authorizations even if an assert statement fails.

Reply:
👍